src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java

```diff
@@ -103,46 +105,63 @@ public String getApiUri() {
+            listener.getLogger().format("Invalid scan credentials, skipping%n");
+            return;
         }
-        if (myself != null && repoOwner.equals(myself.getLogin())) {
-            listener.getLogger().format("Looking up repositories of myself %s%n", repoOwner);
-            for (GHRepository repo : myself.listRepositories()) {
-                if (!repo.getOwnerName().equals(repoOwner)) {
-                    continue; // ignore repos in other orgs when using GHMyself
+
+        if (!github.isAnonymous()) {
+            listener.getLogger().format("Connecting to GitHub using %s%n", CredentialsNameProvider.name(credentials));
+            GHMyself myself = null;
+            try {
+                // Requires an authenticated access
+                myself = github.getMyself();
+            } catch (IOException e) {
+                // Something wrong happened, maybe java.net.ConnectException?
+            }
+            if (myself != null && repoOwner.equals(myself.getLogin())) {
+                listener.getLogger().format("Looking up repositories of myself %s%n%n", repoOwner);
+                for (GHRepository repo : myself.listRepositories()) {
+                    if (!repo.getOwnerName().equals(repoOwner)) {
+                        continue; // ignore repos in other orgs when using GHMyself
+                    }
+                    add(listener, observer, repo);
                 }
-                add(listener, observer, repo);
+                return;
             }
-            return;
+        } else {
+            listener.getLogger().format("Connecting to GitHub using anonymous access%n");
+        }
+
+        GHOrganization org = null;
+        try {
+            org = github.getOrganization(repoOwner);
+        } catch (RateLimitExceededException rle) {
+            listener.getLogger().format("%n%s%n%n", rle.getMessage());
+            throw new InterruptedException();
+        } catch (IOException e) {
+            // may be a user... ok to ignore
         }
-        GHOrganization org = github.getOrganization(repoOwner);
         if (org != null && repoOwner.equals(org.getLogin())) {
-            listener.getLogger().format("Looking up repositories of organization %s%n", repoOwner);
+            listener.getLogger().format("Looking up repositories of organization %s%n%n", repoOwner);
             for (GHRepository repo : org.listRepositories()) {
                 add(listener, observer, repo);
             }
             return;
         }
+
         GHUser user = null;
         try {
             user = github.getUser(repoOwner);
+        } catch (RateLimitExceededException rle) {
+            listener.getLogger().format("%n%s%n%n", rle.getMessage());
+            throw new InterruptedException();
         } catch (IOException e) {
-            // may be organization... ok to ignore
+            // Something wrong happened, maybe java.net.ConnectException?
```

Comment:
:ant: Not part of this PR per se, but anyway: should log such things at least at `FINE`, so we can do diagnosis of problems in the field with a custom logger.